src/utils/events.ts

```diff
@@ -42,13 +42,13 @@ export const eventListenerGenerator = async (
     client = await getCachedColonyClient(contractAddress);
   }
 
-  const filter: { topics: Array<string | null>; address?: string } = {
+  const filter: Filter = {
     topics: [utils.id(eventSignature)],
   };
 
   if (clientType === ClientType.TokenClient) {
     filter.topics = [
-      ...filter.topics,
+      ...(filter.topics ?? []),
```

Comment:
nitpick: this looks a little odd given we know topics is an array here (line 46). But I suppose Filter is telling us that topics might be undefined. 

Maybe this is a good use case for the non-null assertion `!`?

Reply:
So the reason for that is because `filter` in line 45 is defined as of type `Filter`, which says the `topics` field can be undefined/null. At the same time, as you pointed out we know it's definitely not undefined.

The best solution would be to use the `satisfies` operator (https://devblogs.microsoft.com/typescript/announcing-typescript-4-9-beta/#the-satisfies-operator) introduced in TS 4.9 (we're on 4.8.x). It allows you to define the constraint for a value (it has to match the `Filter` type), but at the same time allows TS to infer the more precise type (since we're defining the `topics`, they can't be undefined/null):

```js
const filter = {
  topics: [utils.id(eventSignature)],
} satisfies Filter;
```

Until we upgrade, I think redundancy is safer than assertion (if I used `!` operator, changing topics to null in line 46 would not trigger a compiler error but it would lead to a runtime TypeError).